src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs

```diff
@@ -43,7 +44,7 @@ protected virtual Context CreateContext(ITemplateProvider templateProvider, IDic
                 outerScope: new Hash(),
                 registers: Hash.FromDictionary(new Dictionary<string, object> { { "file_system", templateProvider.GetTemplateFileSystem() } }),
                 errorsOutputMode: ErrorsOutputMode.Rethrow,
-                maxIterations: 0,
+                maxIterations: _maxIterations,
```

Comment:
Will this configuration effect for the following template?

```liquid
{
    {% for i in (1..10000) -%}
        {% for j in (1..10000) -%}
            "value": [{% for k in (1..10000) -%}{{ k }},{% endfor -%}]
        {% endfor -%}
    {% endfor -%}
}
```

Reply:
No, the limitation only works for single for-loop. An unit test is added to show this.

Reply:
Besides, if the nested for-loops take too much time, it should be killed by timeout setting, which is set by FHIR Server.